pkg/operator/apiservicecontroller/apiservice_controller.go

```diff
@@ -230,3 +261,106 @@ func (c *APIServiceController) eventHandler() cache.ResourceEventHandler {
 		DeleteFunc: func(obj interface{}) { c.queue.Add(workQueueKey) },
 	}
 }
+
+// getAPIServicesToManage gets the desired list of API Services that will be managed by this operator
+// note that some services might be managed by an external operators/servers
+func (c *APIServiceController) getAPIServicesToManage() ([]*apiregistrationv1.APIService, error) {
+	if externalOperatorPrecondition := c.externalOperatorPreconditionFunc(); externalOperatorPrecondition != nil {
+		return nil, externalOperatorPrecondition
+	}
+
+	filteredApiServices := []*apiregistrationv1.APIService{}
+
+	// the c.initialAPIServices list is authoritative
+	// it can compete with oauth-server in case of errors but it could save us on downgrade (the current coao and an old cao)
+	for _, apiService := range c.initialAPIServices {
+		if c.isAPIServicePotentiallyManagedByExternalServer(apiService) && c.isAPIServiceAnnotatedByExternalServer(apiService) {
+			continue
+		}
+		filteredApiServices = append(filteredApiServices, apiService)
+	}
+
+	{
+		added, removed := diffAPIServices(c.managedAPIServices, filteredApiServices)
+		if len(added) > 0 {
+			c.eventRecorder.Eventf("OAuthAPIResourcesManagement", "The API Services for the following GVs %v will be managed by this operator (CAS-O)", added)
```

Comment:
This will have the following format `The API Services for the following GVs [v1.oauth.openshift.io v1.user.openshift.io] will be managed by an external operator/server`, do we want to make it pretty?